### PR TITLE
adding support so that the module source can be a local path

### PR DESCRIPTION
**Pull request by silverbp**
In some pipelines artifacts are created as docker images from source and decoupled from source. If you make this modification to the setup.sh script it supports the ability for the module to be local (baked into the image). You can then create a docker image with the terraform image as the base and add your terraform code to a defined path like '/src'.

The only issue I think I have is that I'd like to have the entire source directory accessible even if it's running the create_file module.

Use case is if you have shared modules in the same source/directory..

example directory structure..

/src/modules/shared_module_1
/src/modules/shared_module_2
/src/modules/shared_module_3
/src/create_file

All of these files would be checked into source somewhere, but that source upon commit would generate a docker image.

We already have a pipeline that allows us to release/promote docker images as an artifact from nonprod envs to prod envs, this process works better for us than your traditional gitops where everything is tied directly to source.

It also allows for acquiring other companies that might be on SVN or TFS (you'd eventually want to migrate them to git) 
where it doesn't prohibit you from getting the applications into the pipeline since the universal artifact is a docker image which you can create from any source control system.

```
apiVersion: tf.isaaguilar.com/v1alpha2
kind: Terraform
metadata:
  name: hello-tfo
  namespace: "default"
spec:
  terraformVersion: 1.3.0-beta1
  terraformModule:
    source: /src/create_file

  images:
    terraform:
      image: some-customer-terraform-image:1.0.0
  backend: |-
    terraform {
      backend "kubernetes" {
        secret_suffix    = "hello-tfo"
        in_cluster_config  = true
        namespace = "default"
      }
    }

  keepCompletedPods: true
  ignoreDelete: true
```

### CODE REVIEW

**Comment by isaaguilar** [comment]
@silverbp  that's great. I like it. The issue I see is that `terraformModule.source: /src/foo` will translate to the environment variable as `file:///src/foo`. 

In actuality, I think that `terraformModule.source` is in a way broken because I intended to allow defining the source as a terraform registry address in addition to a url (as it does today).  But I never had a use case so I dropped that feature. 

Perhaps you can make some tweaks to account for `file://` in the env var at the setup.sh level and it should be good to go.

**Comment by silverbp** [comment]
does that handle the ability to copy the entire source into the directory but run a module in that directory in case the module in a directory is referencing another module via relative path in the same folder?

File: setup.sh [hunk 1]
```diff
@@ -48,6 +48,15 @@ if [[ ! -s "$TFO_MAIN_MODULE_ADDONS/inline-module.tf" ]]; then
                 jq -r --arg key "$key" '.data[$key]' <<< $configmap_json > "${TFO_MAIN_MODULE}/${key}"
             done
         fi
+    # Check if this is a source directory instead of a git repo
+    elif [[ $TFO_MAIN_MODULE_REPO == file://* ]]; then
+      local_module_path=${TFO_MAIN_MODULE_REPO#"file://"}
```
Comment (silverbp) [inline setup.sh]: to reference the root you'll have to do `file:///etc/src/blah`

File: setup.sh [hunk 2]
```diff
@@ -48,6 +48,15 @@ if [[ ! -s "$TFO_MAIN_MODULE_ADDONS/inline-module.tf" ]]; then
                 jq -r --arg key "$key" '.data[$key]' <<< $configmap_json > "${TFO_MAIN_MODULE}/${key}"
             done
         fi
+    # Check if this is a source directory instead of a git repo
+    elif [[ $TFO_MAIN_MODULE_REPO == file://* ]]; then
```
Comment (isaaguilar) [inline setup.sh]: In the event of spaces in the var, quote all vars else there is unexpected results. 
- `"$TFO_MAIN_MODULE_REPO"`
- `"${TFO_MAIN_MODULE_REPO#"file://"}"`
- `"$local_module_path"`

Other than this, it looks good.

**Comment by silverbp** [comment]
@isaaguilar does that handle the ability to copy the entire source into the directory but run a module in that directory in case the module in a directory is referencing another module via relative path in the same folder?

**Comment by isaaguilar** [comment]
If `$source`  contains the module that contains relative paths, and `$source` also contains the referenced relative paths, it should work. If the relative paths are not contained in the `$source` directory, it might not work.

**Review by isaaguilar — APPROVED** [review]
Look good

**Pull request merged**